applications/services/storage/storage_processing.c

```diff
@@ -520,45 +520,26 @@ static FS_Error storage_process_sd_status(Storage* app) {
 
 /******************** Aliases processing *******************/
 
-void storage_process_alias(
-    Storage* app,
-    FuriString* path,
-    FuriThreadId thread_id,
-    bool create_folders) {
-    if(furi_string_start_with(path, STORAGE_APP_DATA_PATH_PREFIX)) {
-        FuriString* apps_data_path_with_appsid = furi_string_alloc_set(APPS_DATA_PATH "/");
-        furi_string_cat(apps_data_path_with_appsid, furi_thread_get_appid(thread_id));
-
-        // "/data" -> "/ext/apps_data/appsid"
-        furi_string_replace_at(
-            path,
-            0,
-            strlen(STORAGE_APP_DATA_PATH_PREFIX),
-            furi_string_get_cstr(apps_data_path_with_appsid));
+void storage_process_alias(Storage* app, FuriString* path, bool create_folders) {
+    if(furi_string_start_with(path, STORAGE_APPS_DATA_STEM "/")) {
+        FuriString* apps_data_appid = furi_string_alloc_set(path);
+        // Only create "/ext/apps_data/appid", not any subdir
+        // "/ext/apps_data/appid/whatever" -> "/ext/apps_data/appid"
+        size_t slash =
+            furi_string_search_char(apps_data_appid, '/', strlen(STORAGE_APPS_DATA_STEM "/"));
+        if(slash != FURI_STRING_FAILURE) {
+            furi_string_left(apps_data_appid, slash);
+        }
 
         // Create app data folder if not exists
-        if(create_folders &&
-           storage_process_common_stat(app, apps_data_path_with_appsid, NULL) != FSE_OK) {
-            furi_string_set(apps_data_path_with_appsid, APPS_DATA_PATH);
-            storage_process_common_mkdir(app, apps_data_path_with_appsid);
-            furi_string_cat(apps_data_path_with_appsid, "/");
-            furi_string_cat(apps_data_path_with_appsid, furi_thread_get_appid(thread_id));
-            storage_process_common_mkdir(app, apps_data_path_with_appsid);
+        if(create_folders && storage_process_common_stat(app, apps_data_appid, NULL) != FSE_OK) {
+            FuriString* apps_data = furi_string_alloc_set(STORAGE_APPS_DATA_STEM);
```

Comment:
we're checking existence of one path, but creating a different one

Reply:
it first makes /ext/apps_data here, then 2 lines below it makes /ext/apps_data/appid. it is just ensuring that apps_data exists, previous implementation did this too